libs/dao/src/main/java/com/akto/dao/testing/TestingRunResultDao.java

```diff
@@ -63,35 +68,69 @@ public List<TestingRunResult> fetchLatestTestingRunResult(Bson filters, int limi
                         TestingRunResult.CONFIDENCE_PERCENTAGE,
                         TestingRunResult.START_TIMESTAMP,
                         TestingRunResult.END_TIMESTAMP,
-                        TestingRunResult.TEST_RUN_RESULT_SUMMARY_ID
+                        TestingRunResult.TEST_RUN_RESULT_SUMMARY_ID,
+                        TestingRunResult.TEST_RESULTS + "." + GenericTestResult._CONFIDENCE,
+                        TestingRunResult.TEST_RESULTS + "." + TestResult._ERRORS
                 )
             );
 
         return fetchLatestTestingRunResult(filters, limit, 0, projections);
     }
 
     public List<TestingRunResult> fetchLatestTestingRunResult(Bson filters, int limit, int skip, Bson projections) {
-        MongoCursor<TestingRunResult> cursor = instance.getMCollection().find(filters)
-                .projection(projections)
-                .sort(Sorts.descending("_id"))
-                .skip(skip)
-                .limit(limit)
-                .cursor();
+        List<Bson> pipeline = new ArrayList<>();
+        pipeline.add(Aggregates.match(filters));
+        pipeline.add(Aggregates.project(projections));
+        pipeline.add(Aggregates.sort(Sorts.descending(Constants.ID)));
+        pipeline.add(Aggregates.skip(skip));
```

Comment:
projections or aggregate doesnt have unique key, use TestingRunResult instead of BasicDBObject